programs/mpl-core/src/plugins/oracle.rs

```diff
@@ -57,4 +77,89 @@ pub struct OracleUpdateInfo {
     pub lifecycle_checks: Option<Vec<(HookableLifecycleEvent, ExternalCheckResult)>>,
     /// Optional PDA (derived from Pubkey attached to `ExternalPluginKey`).
     pub pda: Option<ExtraAccount>,
+    /// Optional format for validation results struct used in Oracle account.  Default
+    /// is `ValidationResultsStructFormat::Standard`.
+    pub format: Option<ValidationResultsStructFormat>,
+    /// Optional offset for validation results struct used in Oracle account.  Default
+    /// is `ValidationResultsOffset::NoOffset`.
+    pub results_offset: Option<ValidationResultsOffset>,
+}
+
+/// Format of the validation results struct used in Oracle account.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub enum ValidationResultsStructFormat {
+    /// `Standard` means the Oracle will contain the `OracleValidation` type.
+    Standard,
+    /// `Compact` means the Oracle will contain the `OracleValidationCompact` type.
+    Compact,
+}
+
+/// Offset to where the validation results struct is located in an Oracle account.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub enum ValidationResultsOffset {
+    /// The validation struct is located at the beginning of the account.
+    NoOffset,
+    /// The Oracle is an Anchor account so the validation struct is located after an 8-byte
+    /// account discriminator.
+    Anchor,
+    /// The validation struct is located at the specified offset within the account.
+    Custom(usize),
+}
+
+/// Validation results struct for an Oracle account.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub struct OracleValidation {
+    /// Validation for the the create lifecycle action.
+    pub create: ExternalValidationResult,
+    /// Validation for the transfer lifecycle action.
+    pub transfer: ExternalValidationResult,
+    /// Validation for the burn lifecycle action.
+    pub burn: ExternalValidationResult,
+    /// Validation for the update_plugin lifecycle action.
+    pub update: ExternalValidationResult,
+}
+
+/// Compact validation results struct for an Oracle account.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub struct OracleValidationCompact {
```

Comment:
Similar to the versioning comment above, I'm not sure we need to add compact unless it's asked for. A few thousand lamports probably won't make a difference compared to the 128-byte account overhead. But that's just my 2c.